tests/socket/socket_tests.cpp

```diff
@@ -156,34 +157,402 @@ connection_test(
     receiver_socket.complete_async_receive();
 }
 
-TEST_CASE("connection_test_udp_v4", "[sock_addr_tests]")
+void
+authorization_connect_test(
```

Comment:
I find the name confusing, compared to `connect_authorization_test` or `authorize_connection_test` or `authorize_connect_test`.
Same with `BPF_CGROUP_INET4_AUTH_CONNECT`, why is it not `BPF_CGROUP_INET4_CONNECT_AUTHORIZATION` or `BPF_CGROUP_INET4_AUTHORIZE_CONNECTION` or `BPF_CGROUP_INET4_AUTHORIZE_CONNECT`, any of which would seem more grammatically correct?